src/API/src/allDataFileBuilder.service.ts

```diff
@@ -0,0 +1,56 @@
+const fs = require('fs');
+import { Inject, Injectable } from '@nestjs/common';
+import { handleLastIngestLock } from 'src/ingest/utils/triggerCsvRebuild';
+import { ExportService } from './export/export.service';
+
+interface LastIngest {
+  ingestion: {
+    ingestTime: string;
+    isLocked: boolean;
+  };
+}
+
+@Injectable()
+export class AllDataFileBuilder {
+  private lastIngestTime: LastIngest;
+  constructor(
+    @Inject(ExportService)
+    private readonly exportService: ExportService,
+  ) {
+    this.lastIngestTime = JSON.parse(
+      fs.readFileSync(process.cwd() + '/../../lastIngest.json', {
+        encoding: 'utf8',
+        flag: 'r',
+      }),
+    );
+  }
+
+  async exportAllDataToCsvFile() {
+    await this.exportService.exportOccurrenceDbtoCsvFormat();
+  }
+
+  async lastIngestWatch() {
+    const currentIngestTime: LastIngest = JSON.parse(
```

Comment:
You do this a couple of times - can you pull out the JSON.parse... bit into a separate method?